### PR TITLE
Add tests for ObservationService

### PR DESCRIPTION
**Pull request by gino-m**
Contributes to #821.

Also adds `TestData` builder class with reasonable default values.

### CODE REVIEW

File: web-ng/src/testing/test-data.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import { User } from './../app/shared/models/user.model';
+import firebase from 'firebase/app';
+import { Form } from '../app/shared/models/form/form.model';
+import { Role } from './../app/shared/models/role.model';
+
+/**
+ * Shorthand builders with reasonable defaults for use by tests. Tests should
+ * override values that are relevant even if they're the same as the default
+ * values. Defaults values should only be used for states which are not under
+ * test.
```
Comment (gino-m) [inline web-ng/src/testing/test-data.ts:31]: @os-micmec @DaoyuT FYI, how do you feel about this approach? See `newProject()` below for example.

Comment (DaoyuT) [reply web-ng/src/testing/test-data.ts:31]: This is helping if I need 1 project or 1 layer, what if I need multiple? How can we help that use case?

Comment (gino-m) [reply web-ng/src/testing/test-data.ts:31]: You can call `newProject()` multiple times with different args, incl different layers.

Comment (gino-m) [reply web-ng/src/testing/test-data.ts:31]: Note that I haven't added default args to methods besides `newProject()` and `newLayer`() yet but I plan to.

**Comment by gino-m** [comment]
Obsolete.M erged in #902.

**Pull request closed without merging**